### ISSUE TITLE
wrong code of learning rate?

### ISSUE THREAD

**Issue by cltdevelop** [eragonruan/text-detection-ctpn#201]
Hello, thank you for sharing this code.

I find a problem when I read your code as follows:

# optimizer
        lr = tf.Variable(cfg.TRAIN.LEARNING_RATE, trainable=False)
        if cfg.TRAIN.SOLVER == 'Adam':
            opt = tf.train.AdamOptimizer(cfg.TRAIN.LEARNING_RATE)
        elif cfg.TRAIN.SOLVER == 'RMS':
            opt = tf.train.RMSPropOptimizer(cfg.TRAIN.LEARNING_RATE)
        else:
            # lr = tf.Variable(0.0, trainable=False)
            momentum = cfg.TRAIN.MOMENTUM
            opt = tf.train.MomentumOptimizer(lr, momentum)

Is there something wrong in your learning rate setting? If I use Adam optimizer, the learning rate will not change any more.?

**Comment by eragonruan** [comment]
@cltdevelop check [here](https://github.com/eragonruan/text-detection-ctpn/blob/a550223907a08ae737e28c6ffd65faba263d8ea5/lib/fast_rcnn/train.py#L145)

**Comment by cltdevelop** [comment]
I mean that the learning rate is not related with opt if I use Adam or RMS optimizer, even though you run this code: 'tf.assign(lr, lr.eval() * cfg.TRAIN.GAMMA)'. Because you create opt with constant value: 'opt = tf.train.AdamOptimizer(cfg.TRAIN.LEARNING_RATE)'

**Comment by cltdevelop** [comment]
Oh, yes, I understand it. Thank you very much. @JimReno
